src/main/java/lesson2/problem5/Partition.java

```diff
@@ -0,0 +1,43 @@
+package lesson2.problem5;
+
+import lesson2.problem0.Link;
+import lesson2.problem0.LinkedList;
+
+public class Partition {
+
+    public Link partition(Link head, int pivot) {
+        if (head == null ) {
+            return new Link(0);
+        }
+
+        Link before = new Link(0);
+        Link onOrAfter = new Link(0);
+        Link curr1 = before, curr2 = onOrAfter;
```

Comment:
You can use your LinkedList type rather than using Link. You can add a prepend() to that class, if it makes it easier to work with that class.